cpp11test/src/test-doubles.cpp

```diff
@@ -184,7 +185,12 @@ context("doubles-C++") {
 
 #if defined(__APPLE__) && defined(R_VERSION) && R_VERSION >= R_Version(3, 5, 0)
   test_that("writable::doubles(ALTREP_SEXP)") {
-    SEXP x = PROTECT(Rf_coerceVector(R_compact_intrange(1, 5), REALSXP));
+    // ALTREP compact-seq
+    auto seq = cpp11::package("base")[":"];
+    cpp11::sexp range = seq(cpp11::as_sexp(1), cpp11::as_sexp(5));
+
+    SEXP x = PROTECT(Rf_coerceVector(range, REALSXP));
+    expect_true(ALTREP(x));
```

Comment:
It turns out that `R_compact_intrange()` wasn't supposed to be exposed to us, and R core has moved it to `Defn.h` in R 4.2.0 (or maybe 4.2.1, not sure)
https://github.com/wch/r-source/commit/86b4a0641e889808ce204e28dd8bd19bbd1c8cbb

I've replaced it with a call to `1:5`, which does make an ALTREP object that we can use here, and seemed like a reasonable way to make one for now (with the alternative being that we create a full altrep dummy class in cpp11test to test with)

Corresponds to this commit https://github.com/r-lib/cpp11/pull/279/commits/29cc8339811fe1eb91f2d7a4886111d9f881dd33

Reply:
Makes sense, wonder if it is more readable to use `seq` rather than `:`, though the former is a S3 generic, so not quite the same thing, I don't think it will ever matter in practice.